dpnp/random/_random.pyx

```diff
@@ -111,6 +113,35 @@ cpdef dparray dpnp_exponential(double beta, size):
     return result
 
 
+cpdef dparray dpnp_negative_binomial(double a, double p, size):
+    """
+    Returns an array populated with samples from negative binomial distribution.
+
+    `negative_binomial` generates a matrix filled with random floats sampled from a
+    univariate negative binomial distribution for a given parameter of the distribution
+    `a` and success probability `p` of a single trial.
+
+    """
+
+    dtype = numpy.int32
+
+    # convert string type names (dparray.dtype) to C enum DPNPFuncType
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(dtype)
+
+    # get the FPTR data structure
+    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_NEGATIVE_BINOMIAL, param1_type, param1_type)
+
+    result_type = dpnp_DPNPFuncType_to_dtype( < size_t > kernel_data.return_type)
+    # ceate result array with type given by FPTR data
+    cdef dparray result = dparray(size, dtype=result_type)
+
+    cdef fptr_custom_rng_negative_binomial_c_1out_t func = <fptr_custom_rng_negative_binomial_c_1out_t > kernel_data.ptr
+    # call FPTR function
+    func(result.get_data(), a, p, result.size)
```

Comment:
Have you checked distribution parameters?

Reply:
Added check and covered by tests